### PR TITLE
fix: border size getters when implicit borders are present

### PR DESCRIPTION
**Pull request by meowgorithm**
When a border style is set and no borders have been explicitly enabled or disabled, a border is implicitly applied to all sides of the block. Prior to this fix border size getters would incorrectly return 0 when an implicit border is set. This revision corrects that behavior.

In other words:

```go
s := lipgloss.NewStyle.Border(lipgloss.NormalBorder)

// Before
fmt.Println(s.GetBorderLeftSize()) // 0

// After
fmt.Println(s.GetBorderLeftSize()) // 1
```

### CODE REVIEW

**Comment by meowgorithm** [comment]
@bashbunni Just a heads up about this one. Technically it's a behavioral change but I think it _is_ the right approach here. I plan on merging it today.

I encountered the need for the fix in https://github.com/charmbracelet/bubbletea/pull/1204.

**Comment by bashbunni** [comment]
Hey @meowgorithm just a heads up, this is super similar to another open PR https://github.com/charmbracelet/lipgloss/pull/282/files

The main difference being naming + I like your `GetBorderXSize()` implementations better.

File: get.go [hunk 1]
```diff
@@ -519,6 +519,23 @@ func (s Style) getBorderStyle() Border {
...
+// on or off.
+func (s Style) implicitBorders() bool {
+	var (
+		borderStyle = s.getBorderStyle()
+		topSet      = s.isSet(borderTopKey)
+		rightSet    = s.isSet(borderRightKey)
+		bottomSet   = s.isSet(borderBottomKey)
+		leftSet     = s.isSet(borderLeftKey)
+	)
+	if borderStyle != noBorder && !(topSet || rightSet || bottomSet || leftSet) {
```
Comment (bashbunni) [inline get.go:533]: I think you can remove this `if` condition and just return
```suggestion
	return borderStyle != noBorder && !(topSet || rightSet || bottomSet || leftSet)
```

Comment (meowgorithm) [reply get.go:533]: Good catch!

File: get.go [hunk 2]
```diff
@@ -519,6 +519,23 @@ func (s Style) getBorderStyle() Border {
...
+		borderStyle = s.getBorderStyle()
+		topSet      = s.isSet(borderTopKey)
+		rightSet    = s.isSet(borderRightKey)
+		bottomSet   = s.isSet(borderBottomKey)
+		leftSet     = s.isSet(borderLeftKey)
+	)
+	return borderStyle != noBorder && !(topSet || rightSet || bottomSet || leftSet)
+		return true
+	}
+	return false
```
Comment (aymanbagabas) [inline get.go:536]:
```suggestion
```

Comment (bashbunni) [reply get.go]: Thank you!

**Pull request merged**